test/officerManager.js

```diff
@@ -0,0 +1,336 @@
+        major: 'major',
+        token: token
+      }
+      chai
+        .request(app)
+        .post('/api/officerManager/submit')
+        .send(form)
+        .then(function (res) {
+          expect(res).to.have.status(200)
+
+          done()
+        })
+        .catch(err => {
+          throw err
+        })
+    })
+  })
+
+  describe('/POST GetForm', () => {
+    it('Should return an object of all forms', done => {
+      const form = { token: token }
+      chai
+        .request(app)
+        .post('/api/officerManager/GetForm')
+        .send(form)
+        .then(function (res) {
+          expect(res).to.have.status(200)
+          res.body.should.be.a('array')
+
+          done()
+        })
+        .catch(err => {
+          throw err
+        })
+    })
+
+    it('Should return an object of only querried parameter', done => {
+      const form = {
+        email: 'test@test.com',
+        token: token
+      }
+      chai
+        .request(app)
+        .post('/api/officerManager/GetForm')
+        .send(form)
+        .then(function (res) {
+          expect(res).to.have.status(200)
+          res.body.should.be.a('array')
+          // make sure all child has querries parameter
+          res.body.forEach(obj => {
+            if (obj.email !== 'test@test.com') {
+              throw new Error(
+                'failed query, one or more object did not have query parameter'
+              )
+            }
+          })
+
+          done()
+        })
+        .catch(err => {
+          throw err
+        })
+    })
+  })
+
+  describe('/POST edit', () => {
+    it('Should return statusCode 500 if no token is passed in', done => {
+      const form = {
+        name: 'pinkUnicorn'
+      }
+      chai
+        .request(app)
+        .post('/api/officerManager/edit')
+        .send(form)
+        .then(function (res) {
+          expect(res).to.not.have.status(200)
+
+          done()
+        })
+        .catch(err => {
+          throw err
+        })
+    })
+
+    it('Should return statusCode 401 if an invalid token was passed in', done => {
+      const form = {
+        token: 'Invalid token',
+        email: 'test@test.com'
+      }
+      chai
+        .request(app)
+        .post('/api/officerManager/edit')
+        .send(form)
+        .then(function (res) {
+          expect(res).to.not.have.status(200)
+
+          done()
+        })
+        .catch(err => {
+          throw err
+        })
+    })
+
+    it('Should return statusCode 404 if no form was found', done => {
+      const form = {
+        token: token,
+        email: 'invalid-email'
+      }
+      chai
+        .request(app)
+        .post('/api/officerManager/edit')
+        .send(form)
+        .then(function (res) {
+          expect(res).to.have.status(404)
+
+          done()
+        })
+        .catch(err => {
+          throw err
+        })
+    })
+
+    it('Should return statusCode 200 and a message if a form was edited', done => {
```

Comment:
After updating, please verify that the email with this value can now be found. This is done after editing and deleting in the Event API, found [here](https://github.com/SCE-Development/Core-v4/blob/7eb49330115efe0619c97377f7b69382cca2d41d/test/Event.js#L214)